### PR TITLE
set event.fired_at client-side where it belongs

### PR DESCRIPTION
**Pull request by wnadeau**
hmmmm, well shit

### CODE REVIEW

**Comment by wnadeau** [comment]
if we just deployed this code as-in, some events would end up not having a fired_at

unless we brought the site down & emptied the queues, then deployed the new code

**Comment by wnadeau** [comment]
so I’d ask you all: how tolerant are we of this data being inconsistent? 

Should we rename historical `fired_at` keys to `processed_at` and add `processed_at` back in as a "feature" in this PR?

If no consumers really care about the impact of this, perhaps it's not worth the effort.

File: lib/reactor/event.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def perform(name, data)
     end
 
     def publish(name, data = {})
-      message = new(data.merge(event: name, uuid: SecureRandom.uuid))
+      message = new(data.merge(event: name, uuid: SecureRandom.uuid, fired_at: Time.current))
```
Comment (jeff-hired) [inline lib/reactor/event.rb]: For events coming from either the browser or mobile phone, they should send a timestamp generated by the client. Then you have to do a check to ensure the timestamp seems accurate and possibly adjust it. See: https://amplitude.com/blog/2014/07/21/dont-trust-client-data/

Comment (wnadeau) [inline lib/reactor/event.rb]: Hmmmm, from the context of a standard rails web request, this is the client. (All those server-side `action_event` calls)

But, from the context of a JS/iOS/Android activity, it's not the client and may even overwrite the 'fired_at' that a client could provide....

Perhaps this shouldn't be a feature of reactor at all!

Comment (wnadeau) [reply lib/reactor/event.rb]: > Perhaps this shouldn't be a feature of reactor at all!

Ding ding ding. We have a winner!

**Comment by wnadeau** [comment]
I followed up with @andrew-candela and #data by email, tldr its cool I'm just keeping him in the loop as reactor 1.0 gets ready for deployment

**Comment by wnadeau** [comment]
As I've been preparing this for hired/hired... it's become sadly more clear that we have some underlying /fundamental issues that should block this from getting into a 1.0 release. 😢 (Yes, I'm pretty sure I need to revert this out of master. Ouch.)

Here's my first attempt to explain the WTF that I'm seeing...

It all boils down to how Reactor uses Sidekiq's `perform_at` for 'future-published' events. (which I _did_ pitch removing/re-writing in #21 with [hired/chron](github.com/hired/chron) but with no luck ... so we can add this big WTF to the list of problems our usage of `perform_at` has caused.) Sometimes even foresight isn't enough... and my own career took me away from Reactor for several years... so I'm feeling some pangs of regret at the moment.

It seems we can't move setting `published_at` to `Reactor::Event.publish` (as it is done in this PR) right now because the currently scheduled events are counting on `fired_at` to get set once their Sidekiq job is processed.

We _could_ set a `published_at` and a `processed_at`... but whatever service is listening to the firehose is going need to know that, for these special 'future-publishing events', `published_at` is actually logically equivalent to `scheduled_at`, and a theoretical `processed_at` is logically equivalent to `published_at`.... TOO MANY WTFs.

As such, especially given my upcoming paternity leave, we gotta 🔪  this from 1.0 #70

**Pull request merged**